.editorconfig

```diff
@@ -0,0 +1,116 @@
+# Rules in this file were initially inferred by Visual Studio IntelliCode default codebase
+# You can modify the rules from these initially generated values to suit your own policies
+# You can learn more about editorconfig here: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
+[*.cs]
+
+
+#Core editorconfig formatting - indentation
+
+#use soft tabs (spaces) for indentation
+indent_style = space
+
+#Formatting - new line options
+
+#place catch statements on a new line
+csharp_new_line_before_catch = true
+#require members of anonymous types to be on separate lines
+csharp_new_line_before_members_in_anonymous_types = true
+#require members of object intializers to be on separate lines
+csharp_new_line_before_members_in_object_initializers = true
+#require braces to be on a new line for anonymous_types, methods, control_blocks, types, and object_collection_array_initializers (also known as "Allman" style)
+csharp_new_line_before_open_brace = anonymous_types, methods, control_blocks, types, object_collection_array_initializers
+
+#Formatting - organize using options
+
+#sort System.* using directives alphabetically, and place them before other usings
+dotnet_sort_system_directives_first = true
+
+#Formatting - header template
+
+#add file header template
```

Comment:
I don't mind the file header. but the other rules should be the default out of Visual Studio.

Reply:
@hassanhabib ok, the right click on the solution > add new editor config also exported the other settings.  I have remove everything and only left the two settings on the PR as I have seen them as corrections a few times.